hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -455,6 +459,49 @@ SqlResult execute(IMapDeletePlan plan, List<Object> arguments, long timeout) {
         return arguments;
     }
 
+    /**
+     * Consider
+     * <pre>
+     *  CREATE VIEW v1 AS SELECT * FROM VALUES(1, 1)
+     *  </pre>
+     * <pre>
+     *  CREATE OR REPLACE VIEW v1 AS SELECT 1 FROM VALUES(1, 1)
+     *  </pre>
+     * <p>
+     * We forbid view replacement with incompatible projection
+     * to avoid additional complexity during dependent view processing.
+     * It still doesn't cover replacement in DROP & CREATE VIEW way,
+     * but existing way is better than nothing...
+     */
+    private static void checkProjectsCompatibility(View original, View replacement) {
```

Comment:
It's not enough to compare types, we also must compare names. Consider:
```
create view v as 
select 42 as a, 'foo' as b;

create or replace view
select 42 as b, 'foo' as a;
```

Normally, databases allow adding more columns to views too, they don't allow removing or changing columns.

I think we can quickly fix this, even though it's just a small mitigation, not a fix.